content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md

```diff
@@ -0,0 +1,214 @@
+between pods and containers inside a pod. It is a generalization of the
+persistent volumes API for generic resources. Third-party resource drivers are
+responsible for tracking and allocating resources. Different kinds of
+resources support arbitrary parameters for defining requirements and
+initialization.
+
+## {{% heading "prerequisites" %}}
+
+Kubernetes v{{< skew currentVersion >}} includes cluster-level API support for
+dynamic resource allocation, but it [needs to be
+enabled](#enabling-dynamic-resource-allocation) explicitly.  You also must
+install a resource driver for specific resources that are meant to be managed
+using this API.  If you are not running Kubernetes v{{< skew currentVersion>}},
+check the documentation for that version of Kubernetes.
+
+<!-- body -->
+
+## API
+
+The new `resource.k8s.io/v1alpha1` {{< glossary_tooltip text="API group"
+term_id="api-group" >}} provides four new types:
+
+ResourceClass
+: Defines which resource driver handles a certain kind of
+  resource and provides common parameters for it. ResourceClasses
+  are created by a cluster administrator when installing a resource
+  driver.
+
+ResourceClaim
+: Defines a particular resource instances that is required by a
+  workload. Created by a user (lifecycle managed manually, can be shared
+  between different Pods) or for individual Pods by the control plane based on
+  a ResourceClaimTemplate (automatic lifecycle, typically used by just one
+  Pod).
+
+ResourceClaimTemplate
+: Defines the spec and some meta data for creating
+  ResourceClaims. Created by a user when deploying a workload.
+
+PodScheduling
+: Used internally by the control plane and resource drivers
+  to coordinate pod scheduling when ResourceClaims need to be allocated
+  for a Pod.
+
+Parameters for ResourceClass and ResourceClaim are stored in separate objects,
+typically using the type defined by a {{< glossary_tooltip
+term_id="CustomResourceDefinition" text="CRDs" >}} that was created when
+installing a resource driver.
+
+The `core/v1` `PodSpec` defines ResourceClaims that are needed for a Pod in a new
+`resourceClaims` field. Entries in that list reference either a ResourceClaim
+or a ResourceClaimTemplate. When referencing a ResourceClaim, all Pods using
+this PodSpec (for example, inside a Deployment or StatefulSet) share the same
+ResourceClaim instance. When referencing a ResourceClaimTemplate, each Pod gets
+its own instance.
+
+The `resources.claims` list for container resources define whether a container gets
+access to these resource instances, which makes it possible to share resources
+between one or more containers.
+
+Here is an example for a fictional resource driver. Two ResourceClaim objects
+will get created for this Pod and each container gets access to one of them.
+
+```yaml
+apiVersion: resource.k8s.io/v1alpha1
+kind: ResourceClass
+name: resource.example.com
+driverName: resource-driver.example.com
+---
+apiVersion: cats.resource-driver.example.com/v1
```

Comment:
Does it make more sense to call this `dra.resource-driver.example.com/v1`?

That is the pattern I follow with the NVIDIA GPU resource driver (i.e. `dra.gpu.nvidia.com/v1`). I then get specific in the CRD types that are supported by this API, e.g. (`GpuClaimParameters`, `MigDeviceClaimParameters`, etc.).

Reply:
Isn't "dra" redundant? We were trying not to expose that acronym to users because it's non-obvious, even in a page about "Dynamic Resource Allocation": the acronym never gets introduced...

Reply:
We can get rid of it. I have actually started recommending we use `resource.nvidia.com` to mimic the `resource.k8s.io` API group. We need 'some' prefix on this, and just using `gpu` or `net` feels wrong somehow. I'm happy with `resource` though, if you are.

Reply:
> We need 'some' prefix on this

Why?

So you want `resource.resource-driver.example.com/v1`?

That looks redundant to me.

Reply:
>> We need 'some' prefix on this

> Why?

I just meant it shouldn't be simply `example.com` without `<some-prefix>.example.com`.  I agree that `resource.resource-driver.example.com/v1` is too long (and redundant).

Reply:
It's `cats.resource.example.com/v1` now.